engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/EngineJobExecutionActor.scala

```diff
@@ -263,47 +263,94 @@ class EngineJobExecutionActor(replyTo: ActorRef,
       // Can't write hashes for this job, but continue to wait for the copy response.
       stay using data.copy(hashes = Option(Failure(t)))
   }
+  
```

Comment:
all that follows is just a logical reorganization of the various EJEA cases or are there new cases or ?

Reply:
There's 1 new case that will wait for the hashes if `writeToCache` is on and the job failed. Before it would stop without waiting for the hashes. All the rest is the same, just more explicitly stated, specifically for the aborted case.